packages/pulp-docker/pulp-docker.spec

```diff
@@ -1,14 +1,20 @@
 %{!?python_sitelib: %global python_sitelib %(%{__python} -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())")}
 %{!?python_sitearch: %global python_sitearch %(%{__python} -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(1))")}
 
+# ---- archive related macros ----
+
+%define git_tag 3.1.2
+%define srcname pulp_docker
+
+
 Name: pulp-docker
-Version: 3.2.0
-Release: 0.1.alpha%{?dist}
+Version: 3.1.2
+Release: 1%{?dist}
 Summary: Support for Docker layers in the Pulp platform
 Group: Development/Languages
 License: GPLv2
 URL: http://pulpproject.org
-Source0: https://github.com/%{name}/%{name}/archive/%{name}-%{version}.tar.gz
+Source0: https://github.com/pulp/%{srcname}/archive/%{git_tag}/%{srcname}-%{git_tag}.tar.gz
```

Comment:
I've seen others use the ```Source0: https://url#/%{name}-%{version}.tar.gz``` which means you don't need to modify ```%setup```

Reply:
srcname and Name are not equal, srcname has the '_'

Reply:
It would then become ```Source0: https://github.com/pulp/%{srcname}/archive/%{git_tag}/%{srcname}-%{git_tag}.tar.gz#/%{name}-%{version}.tar.gz```. See https://github.com/theforeman/foreman-packaging/blob/648a9c57cd2d3cdbe28fdd9bbea0c1ed486c7e5e/packages/katello/katello-host-tools/katello-host-tools.spec#L8

Reply:
I tested this change.

The underlying issue is, while the tarball does get renamed,the underlying extraction directory isn't.  Therefore, it still needs to be able to change dir to srcname-gittag, facilitating the need for the -n on setup.  The tar ball provided by github extracts to the project's name-tag folder.  Pulp's rpm names and github project names differ by a character, and thus srcname is needed.

Reply:
Sad :( In that case I wonder if ```#/%{srcname}-%{version}.tar.gz``` would work and keep ```%setup``` minimal.